src/components/Drawer.js

```diff
@@ -0,0 +1,62 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const Button = require('../components/Button');
+
+const Drawer = React.createClass({
+  propTypes: {
+    onClose: React.PropTypes.func.isRequired
+  },
+
+  componentDidMount () {
+    const el = this._container;
+    const transition = { right: 0 };
+    const options = {
+      duration: 500
+    };
+
+    Velocity(el, transition, options);
+  },
+
+  componentWillUnmount () {
+    console.log('component unmounted');
+  },
+
+  _handleCloseClick () {
+    const el = this._container;
+    const transition = { right: -800 };
+    const options = {
+      duration: 500,
+      complete: this.props.onClose
```

Comment:
thanks @tumentumurchudur, I especially like this because it allows the parent component to add whatever it needs to handle to the onClose call, subsequent animations to surrounding components, event handler cleanup or whatever it needs to do without polluting the child component.